pkg/processor/build/runtime/runtime.go

```diff
@@ -75,17 +75,11 @@ func NewAbstractRuntime(logger nuclio.Logger,
 	}
 
 	// create a docker client
-	newRuntime.DockerClient, err = dockerclient.NewShellClient(newRuntime.Logger)
+	newRuntime.DockerClient, err = dockerclient.NewShellClient(newRuntime.Logger, nil)
 	if err != nil {
 		return nil, errors.Wrap(err, "Failed to create docker client")
 	}
 
-	// set cmdrunner
-	newRuntime.CmdRunner, err = cmdrunner.NewShellRunner(newRuntime.Logger)
```

Comment:
Why remove it? I imagine some runtimes will want to make use of it in the future

Reply:
It's instantiated inside "NewShellClient()", why override it with a new instance of the same object?

Reply:
`ShellRunner` is used to run any command, and the runtime should have the ability to do so. It is also instantiated from `dockerclient.NewShellClient`, but that happens implicitly. Right now `runtime.CmdRunner` is `nil`, but even pointing it to use the dockerclient's seems unintuitive. IMO, the correct way to do this here is either:
- create the `cmdrunner.ShellRunner` first and also feed it to the `dockerclient.ShellClient`
- keep separation by creating the `cmdrunner.ShellRunner` explicitly here, and allowing the `dockerclient.ShellClient` to create its own in the constructor.